### DISCUSSION TITLE
loadEnv inside of workspace

### DISCUSSION THREAD

**Discussion by JamesGreenaway** [vitest-dev/vitest#5003] [Q&A]
I want to load all the environment variables inside a project without a VITE prefix. 
I can do this at a project level by adding:

```js
import { loadEnv } from "vite";
import { defineProject } from "vitest/config";

export default defineProject({
  test: {
    env: loadEnv("", process.cwd(), ""),
  },
});
```

This loads my `.env` file as expected i.e. all the variables including all those that are not prefixed by VITE.

However, when I run Vitest from the root of the monorepo it doesn't load the env vars - they return as `undefined`. Nothing happens either, if I add the `loadEnv` function to my workspace config as an `env` property and extend the path to the correct location like so: 

```env: loadEnv("", `${process.cwd()}/apps/my-app`, "")```

How can I have un-prefixed environment variables inside a workspace config on a per-project basis?

I am able to get the expected result, if I add a `vitest.config.ts` at the root like so:

```js
import { loadEnv } from "vite";
import { defineProject } from "vitest/config";

export default defineProject({
  test: {
    env: loadEnv("", `${process.cwd()}/apps/my-app`, ""),
  },
});
```
Unfortunately this prevents me from loading any more than a single app's environment variables as it is not project specific. I can't seem rename the file and add it to the workspace config manually either, it must be `vitest.config.ts` or the workspace config will not see it?!

**Comment by hi-ogawa** [comment]
Thanks for the report. It sounds like a bug. I would appreciate if you can provide a minimal reproduction so that it's easier to investigate for maintainers.

**Comment by hi-ogawa** [comment]
To make sure we're looking at the same issue, could you confirm `test.env: { ... }` config is not working for workspace project regardless of `loadEnv` utility? This is what I see:

https://stackblitz.com/edit/github-2srz39?file=packages%2Flib1%2Fvite.config.ts


```sh
$ npx vitest run

 FAIL  |lib1| test/double.test.ts > env
AssertionError: expected undefined to be 'hello' // Object.is equality

- Expected: 
"hello"

+ Received: 
undefined

 ❯ eval test/double.test.ts:8:35
      6| 
      7| test('env', () => {
      8|   expect(process.env['SOME_ENV']).toBe('hello');
       |                                   ^
      9| });
     10| 

$ npx vitest run --root packages/lib1
...

 ✓ test/double.test.ts (2)
   ✓ double
   ✓ env
```

**Reply by JamesGreenaway** [reply]
yes correct I see the same issue as you, note that if I add a vite.config at the root of the workspace i can point the loadEnv to the package I want and add any variables inside a .env to the process. like so:

[https://stackblitz.com/edit/github-2srz39-ppdwwj?file=vite.config.ts](https://stackblitz.com/edit/github-2srz39-ppdwwj?file=vite.config.ts)

**Reply by hi-ogawa** [reply]
Hey, thanks for confirming. I created a dedicated issue in https://github.com/vitest-dev/vitest/issues/5016
I think current workaround is to setup `process.env` manually on your own `setupFiles`. The idea would be something like this:
https://stackblitz.com/edit/github-2srz39-zjktpk?file=packages%2Flib1%2Ftest%2Fsetup.ts
